src/osvr/ClientKit/DisplayC.cpp

```diff
@@ -130,10 +134,35 @@ OSVR_ReturnCode osvrClientFreeDisplay(OSVR_DisplayConfig disp) {
     return freed ? OSVR_RETURN_SUCCESS : OSVR_RETURN_FAILURE;
 }
 
+OSVR_ReturnCode osvrClientCheckDisplayStartup(OSVR_DisplayConfig disp) {
+    OSVR_VALIDATE_DISPLAY_CONFIG;
+    return disp->cfg->isStartupComplete() ? OSVR_RETURN_SUCCESS
+                                          : OSVR_RETURN_FAILURE;
+}
+
 OSVR_ReturnCode
-osvrClientCheckDisplayStartup(OSVR_DisplayConfig disp) {
+osvrClientGetNumDisplayInputs(OSVR_DisplayConfig disp,
+                              OSVR_DisplayInputCount *numDisplayInputs) {
+
+    OSVR_VALIDATE_DISPLAY_CONFIG;
+    OSVR_VALIDATE_OUTPUT_PTR(numDisplayInputs, "display inputs count");
+    *numDisplayInputs = disp->cfg->getNumDisplayInputs();
+    return OSVR_RETURN_SUCCESS;
+}
+
+OSVR_ReturnCode osvrClientGetDisplayDimensions(
+    OSVR_DisplayConfig disp, OSVR_DisplayInputCount displayInputIndex,
+    OSVR_DisplayDimension *width, OSVR_DisplayDimension *height) {
+
     OSVR_VALIDATE_DISPLAY_CONFIG;
-    return disp->cfg->isStartupComplete() ? OSVR_RETURN_SUCCESS : OSVR_RETURN_FAILURE;
+    OSVR_VALIDATE_DISPLAY_INPUT_ID;
+    OSVR_VALIDATE_OUTPUT_PTR(width, "width");
+    OSVR_VALIDATE_OUTPUT_PTR(height, "height");
+    auto viewport =
```

Comment:
This is definitely wrong. See previous comment about display model - should have a DisplayInput object with these properties.